e2e/runner/cypress-runner-run-tests.js

```diff
@@ -2,25 +2,68 @@ const cypress = require("cypress");
 
 const { parseArguments, args } = require("./cypress-runner-utils");
 
-const isOpenMode = args["--open"];
+const DEFAULT_PORT = 4000;
+const getHost = () =>
+  `http://localhost:${process.env.BACKEND_PORT ?? DEFAULT_PORT}`;
 
-const runCypress = async (baseUrl, exitFunction = console.log) => {
-  const defaultConfig = {
-    browser: "chrome",
-    configFile: "e2e/support/cypress.config.js",
-    config: {
-      baseUrl,
-    },
-  };
+const configs = {
+  e2e: async () => {
+    const defaultConfig = {
+      browser: "chrome",
+      configFile: "e2e/support/cypress.config.js",
+      config: {
+        baseUrl: getHost(),
+      },
+      testingType: "e2e",
+      openMode: args["--open"] || process.env.OPEN_UI,
```

Comment:
if we set OPEN_UI=false, it sets a string value to `process.env.OPEN_UI`, so it always is resolved to `true` here. 

`process.env.OPEN_UI === 'true'`